waspc/src/Wasp/Generator/WebAppGenerator/AuthG.hs

```diff
@@ -63,20 +63,38 @@ genAuthForms auth =
     [ genLoginForm auth,
       genSignupForm auth
     ]
+    <++> (if AS.App.Auth.isExternalAuthEnabled auth then genExternalAuth auth else return [])
 
 genLoginForm :: AS.Auth.Auth -> Generator FileDraft
 genLoginForm auth =
-  -- TODO: Logic that says "/" is a default redirect on success is duplicated here and in the function below.
-  --   We should remove that duplication.
   compileTmplToSamePath
     [relfile|auth/forms/Login.js|]
-    ["onAuthSucceededRedirectTo" .= fromMaybe "/" (AS.Auth.onAuthSucceededRedirectTo auth)]
+    ["onAuthSucceededRedirectTo" .= C.getOnAuthSucceededRedirectToOrDefault auth]
 
 genSignupForm :: AS.Auth.Auth -> Generator FileDraft
 genSignupForm auth =
   compileTmplToSamePath
     [relfile|auth/forms/Signup.js|]
-    ["onAuthSucceededRedirectTo" .= fromMaybe "/" (AS.Auth.onAuthSucceededRedirectTo auth)]
+    ["onAuthSucceededRedirectTo" .= C.getOnAuthSucceededRedirectToOrDefault auth]
+
+genExternalAuth :: AS.Auth.Auth -> Generator [FileDraft]
+genExternalAuth auth =
+  sequence
+    [ genOAuthCodeExchange auth
+    ]
+    <++> genSocialLoginButtons auth
+
+genSocialLoginButtons :: AS.Auth.Auth -> Generator [FileDraft]
+genSocialLoginButtons auth =
+  return ([C.mkTmplFd (C.asTmplFile [relfile|src/auth/buttons/Google.js|]) | AS.App.Auth.isGoogleAuthEnabled auth])
```

Comment:
I don't think we need the parentheses here. Don't know why the linter isn't complaining though.

Reply:
Good eyes! You are correct (and a better linter!!) :)